File: crates/laya-parse/tests/fixtures/sample.rb
require "json"

# Wraps configuration.
module Config
  DEFAULTS = { retries: 3 }.freeze

  # Loads settings.
  class Loader
    def initialize(path)
      @path = path
    end

    def load
      raw = File.read(@path)
      JSON.parse(raw)
    end

    def self.default
      new("config.json")
    end
  end
end

def helper(x)
  x * 2
end

File: crates/laya-parse/tests/fixtures/sample.js
const fs = require("fs");

const RETRIES = 3;

// Loads a config file.
function loadConfig(path) {
  const raw = fs.readFileSync(path, "utf8");
  return JSON.parse(raw);
}

class Cache {
  constructor() {
    this.map = new Map();
  }

  get(key) {
    return this.map.get(key);
  }

  set(key, value) {
    this.map.set(key, value);
    return this;
  }
}

const handler = async (req) => {
  const cfg = loadConfig(req.path);
  return cfg;
};

module.exports = { loadConfig, Cache, handler };
